proposals/low-level-struct-improvements.md

```diff
@@ -486,189 +480,120 @@ struct S1
 }
 ```
 
-Misc Notes:
+Detailed Notes:
 - A `ref` field can only be declared inside of a `ref struct` 
 - A `ref` field cannot be declared `static`
-- A `ref` field can only be `ref` assigned in the constructor of the declaring
-type.
-- The reference assembly generation process must preserve the presence of a 
-`ref` field inside a `ref struct` 
+- A `ref` field can only be `ref` assigned 
+    - in the constructor of the declaring type
+    - when the RHS is known to be a heap location
+- The reference assembly generation process must preserve the presence of a `ref` field inside a `ref struct` 
 - A `readonly ref struct` must declare its `ref` fields as `readonly ref`
-- The span safety rules for constructors, fields and assignment must be updated
-as outlined in this document.
-- The span safety rules need to include the definition of `ref` values that 
-"refer to the heap". 
+- The span safety rules for constructors, fields and assignment must be updated as outlined in this document.
+- The span safety rules need to include the definition of `ref` values that "refer to the heap". 
 
-### Provide struct this escape annotation
-The rules for the scope of `this` in a `struct` limit the *ref-safe-to-escape* scope to the current method. That means neither `this`, nor any of its fields can return by reference to the caller.
+### Provide lifetime annotations
```

Comment:
I wonder if we should have another annotation for extremely specialized cases where the method implementor wants to prohibit escape, but its not a capturing escape. For instance, the StackAlloc api I proposed in a hackathon a few months ago.
```
class RuntimeHelpers
{
    static Span<T> StackAlloc<T>(int count) {... }
}
```

Ideally there would be an annotation that allows specifying that the Span<T> returned from StackAlloc is not safe to escape.

OTOH, this may be so incredibly niche that we should just special case the function and move on.

Reply:
I thought about this too. I think it would come down to how many of these functions we ended up creating. If it's only going ot be a small number, say < 5, and they're all effectively runtime intrinsics then possibly we just special case them. If it's more than that though then I'd favor adding an annotation for them.